inspire/base/templates/format/record/Inspire_HTML_detailed_macros.tpl

```diff
@@ -328,4 +328,80 @@
       <a class="btn btn-default pull-right" href="/search?p=refersto:{{record['control_number']}}&cc=HEP" data-toggle="modal">View in Search Results</a>
     </div>
   </div>
+{% endmacro %}
+
+{% macro record_plots() %}
+  {% if record.get('urls') %}
+    {% set plots_counter = [] %}
+    {% for url in record.get('urls') %}
+      {% if url.get('url').endswith(".png") or url.get('url').endswith(".jpg") %}
+          {% do plots_counter.append(1) %}
+      {% endif %}
+    {% endfor %}
+  {% endif %}
+
+{% if plots_counter %}
+  <div id="record-plots">
+    <div id="record-plots-title">Plots ({{ plots_counter | length }})</div>
+    <!-- Slider -->
+    <div class="row">
+      <div class="col-xs-12" id="slider">
+        <!-- Top part of the slider -->
+        <div class="row">
+          <div class="col-sm-8" id="carousel-bounding-box">
+            <div class="carousel slide" id="plotsCarousel">
+              <!-- Carousel items -->
+              <div class="carousel-inner">
+                {% for url in record.get('urls') %}
```

Comment:
@jmartinm I'm trying to figure a way to skip the second for loop, do you have any suggestions?

The first loop (just as the macro starts) is used to detect if the record has plots, then this loop is used to display them.